wrangler-service/src/main/java/io/cdap/wrangler/service/directive/DirectivesHandler.java

```diff
@@ -565,11 +552,11 @@ public void execute(HttpServiceRequest request, HttpServiceResponder responder,
         if (directiveRequest == null) {
           throw new BadRequestException("Request body is empty.");
         }
-        directiveRequest.getRecipe().setPragma(addLoadablePragmaDirectives(namespace, directiveRequest));
+        List<String> directives = new LinkedList<>(directiveRequest.getRecipe().getDirectives());
```

Comment:
Any specific reason to use a linked list? It creates a lot of objects and generally slower than array list. I think even if you do a single `.add(0)` array list would still be faster as it adds just a single copy. Another options is to use `ArrayDeque` that allows to add to the head without extra copy (assuming enough capacity)

Reply:
It was mainly for the add(0). BTW, the list is quite short (list of directives), so using anything won't make much difference